.github/workflows/yocto-build-deploy.yml

```diff
@@ -1131,11 +1136,15 @@ jobs:
 
       - name: Decrypt artifacts
         working-directory: ${{ env.WORKSPACE }}
+        if: |
+          github.event.repository.public &&
+          inputs.sign-image == true
```

Comment:
Rather than check the same conditions we used on upload, we could use the .enc extension only when some artifacts are encrypted, so this step is no-op if there are no files with that extension.

That makes it easier to change the conditions of encryption in the future, and even encrypt just a subset of the files, without it being a global on/off toggle.

Reply:
Do we have a private repo, which is producing a signed image we can test this workflow on?

Reply:
Not yet, but we probably should. Let me open a PR we can use to test this PR.

(we need a public repo with signed images, so I'll get balena-generic ready to go)

In the meantime let's also add a condition to encrypt [private device types](https://github.com/balena-os/balena-yocto-scripts/blob/0684ef25f676baddd6c55a504b575776269588de/.github/workflows/yocto-build-deploy.yml#L172).

So `(inputs.sign-image == true || steps.balena-lib.outputs.is_private == 'true')` which would allow testing on balena-raspberrypi which is public with a mix of private and public device types.

Reply:
If that works we can merge without re-testing signed images IMO. At least then we will have the confidence to enable signed images for testing.